### ISSUE TITLE
JupyterLab Broken after clean reinstall. 

### ISSUE THREAD

**Issue by Carreau** [jupyter/jupyterlab#265]
Hi Guys, 

I'm trying to catch up with JupyterLab after having focused on IPython for a few weeks.

I pulled and launched to see the new UI, which looks really nice, we can see things getting closer to completion, I think it will impress people at SciPy. I though have one small issue [which might be a problem of my system], I don't see any menubar, instead the screen is empty. This is regardless of browser [Chome beta, or FF stable, on OSX], screenshot on firefox: 

![screen shot 2016-07-08 at 15 50 50](https://cloud.githubusercontent.com/assets/335567/16703597/c6030504-4523-11e6-81e3-d7f72cc95f5c.png)

> Sidenote, you might notice that there is a white pixel gap 1px just below the "about" tab on the left border, but that's only on some zoom level and not the default one, so likely a Firefox rounding error, so not important. We might find a workaround though, so leaving the there. 

As it might be my system, I decide to clean all `git clean -xfd`, `git reset --hard master`, nuke the server extensions and configs, and re follow the install instructions. 

All is pretty clear, really nice and easy to follow. 

I think trying to be slightly more verbose at the beginning of `pip install -e .` and remind that install can take a long time might be helpful, I almost though my machine had frozen despite reading the warning. 
I can do that later. Just a reminder for me. 

Though this step crash for me, and I'm unable to figure out why:

```
 ModuleNotFoundError: Module not found: Error: Cannot resolve module 'jquery-ui/themes/smoothness/jquery-ui.min.css' in /Users/bussonniermatthias/dev/jupyterlab/jupyterlab/node_modules/jupyterlab/lib/widgets
```

[full traceback](https://gist.github.com/Carreau/3e8823c227d83293eed83d8df08bf98a)

I ran `npm run build:serverextension` manually which – as one would expect – give the same error. 

here is the content of the jquery-ui installed theme:

```
$ pwd
/Users/bussonniermatthias/dev/jupyterlab/node_modules/jquery-ui/themes
(rootenv)~/dev/jupyterlab/node_modules/jquery-ui/themes master$ tree
.
└── base
    ├── accordion.css
    ├── all.css
    ├── autocomplete.css
    ├── base.css
    ├── button.css
    ├── checkboxradio.css
    ├── controlgroup.css
    ├── core.css
    ├── datepicker.css
    ├── dialog.css
    ├── draggable.css
    ├── images
    │   ├── ui-bg_flat_0_aaaaaa_40x100.png
    │   ├── ui-icons_444444_256x240.png
    │   ├── ui-icons_555555_256x240.png
    │   ├── ui-icons_777620_256x240.png
    │   ├── ui-icons_777777_256x240.png
    │   ├── ui-icons_cc0000_256x240.png
    │   └── ui-icons_ffffff_256x240.png
    ├── menu.css
    ├── progressbar.css
    ├── resizable.css
    ├── selectable.css
    ├── selectmenu.css
    ├── slider.css
    ├── sortable.css
    ├── spinner.css
    ├── tabs.css
    ├── theme.css
    └── tooltip.css

2 directories, 29 files
```

Not sure what i am missing, any help appreciated.

**Comment by ellisonbg** [comment]
The menu bar doesn't have any content yet so what you are seeing there is
correct. Still WIP for other plugins to use the menu plugin to add content
- also we are still working on styling the menu. But looks like there may
be other issues you are seeing too?

On Fri, Jul 8, 2016 at 4:05 PM, Matthias Bussonnier <
notifications@github.com> wrote:

> Hi Guys,
>
> I'm trying to catch up with JupyterLab after having focused on IPython for
> a few weeks.
>
> I pulled and launched to see the new UI, which looks really nice, we can
> see things getting closer to completion, I think it will impress people at
> SciPy. I though have one small issue [which might be a problem of my
> system], I don't see any menubar, instead the screen is empty. This is
> regardless of browser [Chome beta, or FF stable, on OSX], screenshot on
> firefox:
>
> [image: screen shot 2016-07-08 at 15 50 50]
> <https://cloud.githubusercontent.com/assets/335567/16703597/c6030504-4523-11e6-81e3-d7f72cc95f5c.png>
>
> Sidenote, you might notice that there is a white pixel gap 1px just below
> the "about" tab on the left border, but that's only on some zoom level and
> not the default one, so likely a Firefox rounding error, so not important.
> We might find a workaround though, so leaving the there.
>
> As it might be my system, I decide to clean all git clean -xfd, git reset
> --hard master, nuke the server extensions and configs, and re follow the
> install instructions.
>
> All is pretty clear, really nice and easy to follow.
>
> I think trying to be slightly more verbose at the beginning of pip
> install -e . and remind that install can take a long time might be
> helpful, I almost though my machine had frozen despite reading the warning.
> I can do that later. Just a reminder for me.
>
> Though this step crash for me, and I'm unable to figure out why:
>
>  ModuleNotFoundError: Module not found: Error: Cannot resolve module 'jquery-ui/themes/smoothness/jquery-ui.min.css' in /Users/bussonniermatthias/dev/jupyterlab/jupyterlab/node_modules/jupyterlab/lib/widgets
>
> full traceback
> <https://gist.github.com/Carreau/3e8823c227d83293eed83d8df08bf98a>
>
> I ran npm run build:serverextension manually which – as one would expect
> – give the same error.
>
> here is the content of the jquery-ui installed theme:
>
> $ pwd
> /Users/bussonniermatthias/dev/jupyterlab/node_modules/jquery-ui/themes
> (rootenv)~/dev/jupyterlab/node_modules/jquery-ui/themes master$ tree
> .
> └── base
>     ├── accordion.css
>     ├── all.css
>     ├── autocomplete.css
>     ├── base.css
>     ├── button.css
>     ├── checkboxradio.css
>     ├── controlgroup.css
>     ├── core.css
>     ├── datepicker.css
>     ├── dialog.css
>     ├── draggable.css
>     ├── images
>     │   ├── ui-bg_flat_0_aaaaaa_40x100.png
>     │   ├── ui-icons_444444_256x240.png
>     │   ├── ui-icons_555555_256x240.png
>     │   ├── ui-icons_777620_256x240.png
>     │   ├── ui-icons_777777_256x240.png
>     │   ├── ui-icons_cc0000_256x240.png
>     │   └── ui-icons_ffffff_256x240.png
>     ├── menu.css
>     ├── progressbar.css
>     ├── resizable.css
>     ├── selectable.css
>     ├── selectmenu.css
>     ├── slider.css
>     ├── sortable.css
>     ├── spinner.css
>     ├── tabs.css
>     ├── theme.css
>     └── tooltip.css
>
> 2 directories, 29 files
>
> Not sure what i am missing, any help appreciated.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/jupyter/jupyterlab/issues/265>, or mute the thread
> <https://github.com/notifications/unsubscribe/AABr0LOoawEYlbWsi_Eo62bAwkltnH6Hks5qTtfKgaJpZM4JId3B>
> .
>



-- 
Brian E. Granger
Associate Professor of Physics and Data Science
Cal Poly State University, San Luis Obispo
@ellisonbg on Twitter and GitHub
bgranger@calpoly.edu and ellisonbg@gmail.com

**Comment by Carreau** [comment]
Of course writing the issue allowed me to find the reason. 

Jquery-UI 1.12 breaks API with 1.11 [regarding the theme](https://jqueryui.com/upgrade-guide/1.12/#new-default-theme), and was release 6h ago... will need to roll-back.

**Comment by willingc** [comment]
@Carreau Yeah, I think @blink1073 is aware of the jquery thing since it's been bugging travis today.

**Comment by ellisonbg** [comment]
ahh, great catch!

On Fri, Jul 8, 2016 at 4:12 PM, Carol Willing <notifications@github.com>
wrote:

> @Carreau <https://github.com/Carreau> Yeah, I think @blink1073
> <https://github.com/blink1073> is aware of the jquery thing since it's
> been bugging travis today.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/jupyter/jupyterlab/issues/265#issuecomment-231494039>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AABr0NRX_V8ZtP5LMWZkC7r9XcfWk_Poks5qTtlogaJpZM4JId3B>
> .
>



-- 
Brian E. Granger
Associate Professor of Physics and Data Science
Cal Poly State University, San Luis Obispo
@ellisonbg on Twitter and GitHub
bgranger@calpoly.edu and ellisonbg@gmail.com

**Comment by Carreau** [comment]
> The menu bar doesn't have any content yet so what you are seeing there is
> correct. Still WIP for other plugins to use the menu plugin to add content
> - also we are still working on styling the menu

Ok, with the white thing on the far-left I had the impression that I had a menu but out of the screen. 
can I suggest temporarily have a dif saying "menu placeholder" to not confuse newcomer ?

> @Carreau Yeah, I think @blink1073 is aware of the jquery thing since it's been bugging travis today.

Great! Is he trying to fix it, or should we just pin jquery to `<1.12` ?

**Comment by Carreau** [comment]
> But looks like there may
> be other issues you are seeing too?

Yes, I have some suggestions, I'll open separate issues. I want to rebuild Jlab successfully first.

**Comment by willingc** [comment]
@Carreau I think he was done for the day, but will look at it tomorrow.

**Comment by ellisonbg** [comment]
Yes, your suggestion on a menu placeholder is great - one of my students is
working on that in one of his PRs...

On Fri, Jul 8, 2016 at 4:15 PM, Matthias Bussonnier <
notifications@github.com> wrote:

> The menu bar doesn't have any content yet so what you are seeing there is
> correct. Still WIP for other plugins to use the menu plugin to add content
>
>    - also we are still working on styling the menu
>
> Ok, with the white thing on the far-left I had the impression that I had a
> menu but out of the screen.
> can I suggest temporarily have a dif saying "menu placeholder" to not
> confuse newcomer ?
>
> @Carreau <https://github.com/Carreau> Yeah, I think @blink1073
> <https://github.com/blink1073> is aware of the jquery thing since it's
> been bugging travis today.
>
> Great! Is he trying to fix it, or should we just pin jquery to <1.12 ?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/jupyter/jupyterlab/issues/265#issuecomment-231494411>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AABr0B5cFX1UxDx6Z3ux3ucdWQ3EP2hpks5qTtoegaJpZM4JId3B>
> .
>



-- 
Brian E. Granger
Associate Professor of Physics and Data Science
Cal Poly State University, San Luis Obispo
@ellisonbg on Twitter and GitHub
bgranger@calpoly.edu and ellisonbg@gmail.com

**Comment by blink1073** [comment]
Fixed by #267.  Thanks again, @Carreau!
